### PR TITLE
Deprecate old fields that were pointing to users

### LINKED ISSUE #2063: Deprecate old fields that were pointing to users
## Scope & Context

As part of: https://github.com/twentyhq/twenty/issues/1388 we have performed: https://github.com/twentyhq/twenty/pull/1514 which has introduced

on WorkspaceMember model:
```
  authoredActivities  Activity[]    @relation(name: "authoredActivities")
  assignedActivities  Activity[]    @relation(name: "assignedActivities")
  authoredAttachments Attachment[]  @relation(name: "authoredAttachments")
  settings            UserSettings? @relation(fields: [settingsId], references: [id])
  settingsId          String?
  companies Company[]
  comments  Comment[]
```

on Company:
```
workspaceMemberAccountOwner   WorkspaceMember? @relation(fields: [workspaceMemberAccountOwnerId], references: [id], onDelete: SetNull)
  workspaceMemberAccountOwnerId String?
```

on Activity:
```
  workspaceMemberAuthor   WorkspaceMember? @relation(fields: [workspaceMemberAuthorId], references: [id], name: "authoredActivities", onDelete: Cascade)
  workspaceMemberAuthorId String?

  workspaceMemberAssignee   WorkspaceMember? @relation(fields: [workspaceMemberAssigneeId], references: [id], name: "assignedActivities", onDelete: SetNull)
  workspaceMemberAssigneeId String?
```

on Comment:
```
  workspaceMemberAuthor   WorkspaceMember? @relation(fields: [workspaceMemberAuthorId], references: [id], onDelete: Cascade)
  workspaceMemberAuthorId String?
```

and on Attachment:
```
  workspaceMemberAuthor   WorkspaceMember? @relation(fields: [workspaceMemberAuthorId], references: [id], name: "authoredAttachments", onDelete: Cascade)
  workspaceMemberAuthorId String?
```

The data has been backfilled in production.

In this PR: 
- delete the fields pointing to users (example Comment->author and Comment->authorId)
- in another migration (to avoid column renaming with FK issues): rename Comment->workspaceMemberAuthor to Comment->author) and clean the business logic

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Fixes #2063

### CODE REVIEW

**Comment by charlesBochet** [comment]
@gitstart-twenty is this PR ready for review?

File: server/src/database/migrations/20231018054311_rename_workspace_member_fields/migration.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ALTER TABLE "attachments" DROP CONSTRAINT "attachments_workspaceMemberAuthorId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "comments" DROP CONSTRAINT "comments_workspaceMemberAuthorId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "companies" DROP CONSTRAINT "companies_workspaceMemberAccountOwnerId_fkey";
+
+-- AlterTable
+ALTER TABLE "activities" DROP COLUMN "workspaceMemberAssigneeId",
```
Comment (charlesBochet) [inline server/src/database/migrations/20231018054311_rename_workspace_member_fields/migration.sql:27]: we should rename instead of dropping / adding

**Review by charlesBochet — COMMENTED** [review]
Thank you! I believe they are also some areas to clean in the code regarding settings and activities that were supporting both implementations (on users and workspace members). We should be able to simplify a bit!
See changes in: https://github.com/twentyhq/twenty/pull/1514/files

Also, could you please make sure that all behaviors are still working once done?

**Comment by gitstart-twenty** [comment]
> Also, could you please make sure that all behaviors are still working once done?

Noticed we needed a new picker `WorkspaceMemberPicker`, added it

**Comment by charlesBochet** [comment]
> @charlesBochet Did we move `settings` exclusively to `workspaceMember`?

Yes!

File: front/src/modules/ui/theme/hooks/useColorScheme.ts [hunk 2]
```diff
@@ -6,97 +6,126 @@ import {
...
+        //         "column": 3
+        //       }
+        //     ],
+        //     "path": [
+        //       "updateUserSettings"
+        //     ]
+        //   }
+        // ],
+        // "data": null
+        // }
```
Comment (gitstart-twenty) [inline front/src/modules/ui/theme/hooks/useColorScheme.ts:52]: I am getting an error here for some reason, I haven't figured out why

File: front/src/modules/ui/theme/hooks/useColorScheme.ts [hunk 3]
```diff
@@ -4,131 +4,65 @@ import { useRecoilState } from 'recoil';
...
-        //   }
-        // ],
-        // "data": null
-        // }
-        await updateUserSettings({
+        const result = await updateUserSettings({
           variables: {
             data: {
               colorScheme: value,
+              // FIXME: For some reason, when I send a request with only colorScheme, it fails, weird.
```
Comment (gitstart-twenty) [inline front/src/modules/ui/theme/hooks/useColorScheme.ts:26]: This has bugged me out, can't figure out why it's happening

**Comment by charlesBochet** [comment]
I'm extensively testing this one :)

**Comment by gitstart-twenty** [comment]
> I'm extensively testing this one :)

Alright, sir

**Comment by charlesBochet** [comment]
@gitstart-twenty As for the other two PRs, I'm going to close this one too as we will be leveraging the schema migration to take this refactoring into account. Sorry about that, you can mark this as completed on gitstart side

**Pull request closed without merging**